src/activity_actor.cpp

```diff
@@ -1201,11 +1201,7 @@ void consume_activity_actor::finish( player_activity &act, Character & )
     avatar &player_character = get_avatar();
     if( !canceled ) {
         if( consume_loc ) {
-            trinary result = player_character.consume( consume_loc, /*force=*/true );
-            // Parent pockets need to be notified so they can be unsealed as well.
-            if( result != trinary::NONE ) {
-                game::handle_contents_changed( consume_loc );
```

Comment:
Eat won't be opening the parent pockets if this is removed, if I'm not mistaken.

I think the new handler should go here.

Reply:
The code is moved to `player::consume` now (merged with the previous `Pickup::handle_spillable_contents` call).